tests/moduletool/test_install.py

```diff
@@ -780,6 +780,55 @@ def test_project_install_requirement_not_loaded(
     assert message in (rec.message for rec in caplog.records)
 
 
+def test_pip_extra_index_env(
+    tmpdir: py.path.local,
+    modules_v2_dir: str,
+    snippetcompiler_clean,
+) -> None:
+    """
+    Test that PIP_EXTRA_INDEX_URL is not set in the subprocess doing an install_from_index
+    and that it is not changed in the active env.
+
+    The installation fails with an ModuleNotFoundException
+    as the index https://pypi.org/simple is needed to install lorem~=0.0.1,
+    but it is only present in the active env in PIP_EXTRA_INDEX_URL which is not know by the
+    subprocess doing the pip install.
+    """
+    os.environ["PIP_EXTRA_INDEX_URL"] = "https://pypi.org/simple"
```

Comment:
Setting environment variables should be done using the monkeypatch fixture to make sure the original values get restored at the end of the test case.